src/requirements.cpp

```diff
@@ -333,6 +334,28 @@ void requirement_data::check_consistency()
     }
 }
 
+void requirement_data::finalize( alter_tool_comp_vector &vec )
+{
+    for( auto &list : vec ) {
+        std::vector<tool_comp> new_list;
+        for( auto &comp : list ) {
+            const auto replacements = item_controller->subtype_replacement( comp.type );
+            for( const auto &replaced_type : replacements ) {
+                new_list.push_back( tool_comp( replaced_type, comp.count ) );
+            }
+        }
+
+        list = std::move( new_list );
```

Comment:
I doubt `std::move` will do anything there - the compiler should be able to figure things out itself